endpoints/openrtb2/auction.go

```diff
@@ -1168,87 +1182,88 @@ func validatePmp(pmp *openrtb2.PMP, impIndex int) error {
 	return nil
 }
 
-func (deps *endpointDeps) validateImpExt(imp *openrtb2.Imp, aliases map[string]string, impIndex int, hasStoredResponses bool, storedBidResp stored_responses.ImpBidderStoredResp) []error {
-	errL := []error{}
+func (deps *endpointDeps) validateImpExt(imp *openrtb_ext.ImpWrapper, aliases map[string]string, impIndex int, hasStoredResponses bool, storedBidResp stored_responses.ImpBidderStoredResp) []error {
 	if len(imp.Ext) == 0 {
 		return []error{fmt.Errorf("request.imp[%d].ext is required", impIndex)}
 	}
 
-	var bidderExts map[string]json.RawMessage
-	if err := json.Unmarshal(imp.Ext, &bidderExts); err != nil {
+	impExt, err := imp.GetImpExt()
+	if err != nil {
 		return []error{err}
 	}
 
-	// Prefer bidder params from request.imp.ext.prebid.bidder.BIDDER over request.imp.ext.BIDDER
-	// to avoid confusion beteween prebid specific adapter config and other ext protocols.
-	var extPrebid openrtb_ext.ExtImpPrebid
-	if extPrebidJSON, ok := bidderExts[openrtb_ext.PrebidExtKey]; ok {
-		if err := json.Unmarshal(extPrebidJSON, &extPrebid); err == nil && extPrebid.Bidder != nil {
-			for bidder, ext := range extPrebid.Bidder {
-				if ext == nil {
-					continue
-				}
-				bidderExts[bidder] = ext
+	prebid := impExt.GetOrCreatePrebid()
+	prebidModified := false
+
+	if prebid.Bidder == nil {
+		prebid.Bidder = make(map[string]json.RawMessage)
+	}
+
+	ext := impExt.GetExt()
+	extModified := false
+
+	// promote imp[].ext.BIDDER to newer imp[].ext.prebid.BIDDER location, with the later taking precedence
```

Comment:
The team discussed and we agree that moving everything in `imp[].ext.BIDDER` to `imp[].ext.prebid.bidder.BIDDER` here makes sense and will simplify the downstream logic. We're ok with the resolved request only showing bidder params in the new location, especially since this is now the official location.

Also PBS-Java already does this:
incoming request:
```
"imp": [{
    "ext": {
      "appnexus": {
        "placementId": 12883451
      },
      "prebid": {
          "bidder": {
              "adf": {
                "mid": 12345
            }
          }
      }
    }
  }],
```
resolved request:
```
"imp": [{
    "ext": {
        "prebid": {
            "bidder": {
                "adf": {
                    "mid": 12345
                },
                "appnexus": {
                    "placementId": 12883451
                }
            }
        }
    }
}]
```

Reply:
Also I believe the official location is `imp[].ext.prebid.bidder.BIDDER`, not `imp[].ext.prebid.BIDDER`.

Reply:
Sounds good. Thank you.